### PR TITLE
Issue 57: dev_state/status are not gevent safe

### PR DESCRIPTION
**Pull request by tiagocoutinho**
Need to accept PR #60 first please

### CODE REVIEW

**Comment by vxgmichel** [comment]
LGTM, thanks! PEP8 does complain about an expected blank line before the function definitions though.

**Comment by tiagocoutinho** [comment]
Ok, i will pepify it first

**Comment by tiagocoutinho** [comment]
@vxgmichel, where exactly does it complain for you (file and line)?

File: tango/server.py [hunk 1]
```diff
@@ -413,25 +413,44 @@ def __patch_pipe_methods(tango_device_klass, pipe):
...
-        setattr(klass, "init_device", init_device)
-
-        delete_device_orig = klass.delete_device
-        @functools.wraps(delete_device_orig)
-        def delete_device(self):
-            return get_worker().execute(delete_device_orig, self)
-        setattr(klass, "delete_device", delete_device)
+    init_device_orig = klass.init_device
+    @functools.wraps(init_device_orig)
+    def init_device(self):
```
Comment (vxgmichel) [inline tango/server.py]: Should be:
```
    init_device_orig = klass.init_device

    @functools.wraps(init_device_orig)
    def init_device(self):
        [...]
```

**Comment by tiagocoutinho** [comment]
@vxgmichel, it should be ok now

**Comment by tiagocoutinho** [comment]
Ok, now that @jairomoldes merged #60 it should be ok to merge this one. It is currently in production at ESRF so it should be safe to merge

File: tango/server.py [hunk 2]
```diff
@@ -628,6 +653,15 @@ def delete_device(self):
...
+    def read_attr_hardware(self, attr_list):
+        return LatestDeviceImpl.read_attr_hardware(self, attr_list)
+
+    def dev_state(self):
+        return LatestDeviceImpl.dev_state(self)
+
+    def dev_status(self):
+        return LatestDeviceImpl.dev_status(self)
+
     def get_device_properties(self, ds_class = None):
```
Comment (vxgmichel) [inline tango/server.py]: Ok, do you think it could be any trouble?

**Pull request merged**